doc/developer/design/20250226_postgres_style_explain.md

```diff
@@ -0,0 +1,338 @@
+            ArrangeBy keys=[[#0{o_orderkey}], [#1{o_custkey}]] (columns=9)
+              ReadIndex on=orders pk_orders_orderkey=[delta join lookup] fk_orders_custkey=[delta join lookup] (columns=9)
+            ArrangeBy keys=[[#0{l_orderkey}]] (columns=16)
+              ReadIndex on=lineitem fk_lineitem_orderkey=[delta join lookup] (columns=16)
+
+Used Indexes:
+  - materialize.public.pk_customer_custkey (delta join 1st input (full scan))
+  - materialize.public.pk_orders_orderkey (delta join lookup)
+  - materialize.public.fk_orders_custkey (delta join lookup)
+  - materialize.public.fk_lineitem_orderkey (delta join lookup)
+
+Target cluster: quickstart
+```
+
+New Materialize `EXPLAIN`:
+
+```
+Finish
+  Order by: sum desc nulls_first, o_orderdate
+  -> Project (columns=4)
+     Columns: o_orderkey, sum, o_orderdate, o_shippriority
+     -> Reduce (columns=4)
+        Group key: o_orderkey..=#2o_shippriority
+        Aggregates: sum((l_extendedprice * (1 - l_discount)))
+        -> Project (columns=5)
+           Columns: o_orderkey, o_orderdate, o_shippriority, l_extendedprice, l_discount
+           -> Filter (columns=33)
+              Predicates: (c_mktsegment = "BUILDING") AND (o_orderdate < 1995-03-15) AND (l_shipdate > 1995-03-15)
+              -> Delta Join (columns=33)
+                 Conditions: c_custkey = o_custkey AND o_orderkey = l_orderkey
+                 Pipelines:
+                   %0:customer » %1:orders[#1]KAif » %2:lineitem[#0]KAif
+                   %1:orders » %0:customer[#0]KAef » %2:lineitem[#0]KAif
+                   %2:lineitem » %1:orders[#0]KAif » %0:customer[#0]KAef
+                 -> Arranged (columns=8)
+                    Keys: [c_custkey]
+                    -> Index Scan using pk_customer_custkey on customer (columns=8)
+                       Delta join first input (full scan): pk_customer_custkey
+                 -> Arrange (columns=9)
+                    Keys: [o_orderkey], [o_custkey]
+                    -> Index Scan using pk_orders_orderkey, fk_orders_custkey on orders (columns=9)
+                       Delta join lookup: pk_orders_orderkey, fk_orders_custkey
+                 -> Arrange (columns=16)
+                    Keys: [l_orderkey]
+                    -> Index Scan using fk_lineitem_orderkey on lineitem (columns=16)
+                       Delta join lookup: fk_lineitem_orderkey
+
+Used Indexes:
+  - materialize.public.pk_customer_custkey (delta join 1st input (full scan))
+  - materialize.public.pk_orders_orderkey (delta join lookup)
+  - materialize.public.fk_orders_custkey (delta join lookup)
+  - materialize.public.fk_lineitem_orderkey (delta join lookup)
+
+Target cluster: quickstart
+```
+
+## Alternatives
+
+Should we more radically reduce the AST?
+
+Should we abandon static `EXPLAIN` and encourage `mz_lir_mapping` use?
+
+## Open questions
+
+Should we show `Project`?
+
+Should we show _all_ expressions for `Map` and `Filter`?
```

Comment:
(https://github.com/MaterializeInc/database-issues/issues/8889#issuecomment-2612639970 discusses this in the penultimate paragraph. I'm not sure, but leaning towards yes.)